packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -417,6 +452,23 @@ export class Metric implements IMetric {
       };
     });
   }
+
+  private validateDimensionsMap(dims: DimensionsMap): void {
```

Comment:
I'm not a huge fan of this duplication between `validateDimensionsMap` and `validateDimensions`.

Can we instead call `validateDimensions`, but passing `props.dimensionsMap`? I think the types should work out there. Sure, the `undefined` and `null` checks will be redundant, but that's fine.